src/Network/Minio/PutObject.hs

```diff
@@ -117,94 +112,57 @@ selectPartSizes size = uncurry (List.zip3 [1..]) $
-checkUploadNeeded :: Payload -> PartNumber
-                  -> Map.Map PartNumber ObjectPartInfo
-                  -> Minio (Maybe PartTuple)
-checkUploadNeeded payload n pmap = do
-  (md5hash, pSize) <- case payload of
-    PayloadBS bs -> return (hashMD5 bs, fromIntegral $ B.length bs)
-    PayloadH h off size -> fmap (, size) $
-      hashMD5FromSource $ sourceHandleRange h (Just $ fromIntegral off)
-      (Just $ fromIntegral size)
-  case Map.lookup n pmap of
-    Nothing -> return Nothing
-    Just (ObjectPartInfo _ etag size _) -> return $
-      bool Nothing (Just (n, etag)) $
-      md5hash == encodeUtf8 etag && size == pSize
-
 parallelMultipartUpload :: Bucket -> Object -> FilePath -> Int64
                         -> Minio ETag
 parallelMultipartUpload b o filePath size = do
-  (uidMay, pmap) <- getExistingUpload b o
-
-  -- get a new upload id if needed.
-  uploadId <- maybe (newMultipartUpload b o []) return uidMay
+  -- get a new upload id.
+  uploadId <- newMultipartUpload b o []
 
   let partSizeInfo = selectPartSizes size
 
   -- perform upload with 10 threads
   uploadedPartsE <- limitedMapConcurrently 10
-                    (uploadPart pmap uploadId) partSizeInfo
+                    (uploadPart uploadId) partSizeInfo
 
   -- if there were any errors, rethrow exception.
   mapM_ throwM $ lefts uploadedPartsE
 
   -- if we get here, all parts were successfully uploaded.
   completeMultipartUpload b o uploadId $ rights uploadedPartsE
   where
-    uploadPart pmap uploadId (partNum, offset, sz) =
+    uploadPart uploadId (partNum, offset, sz) =
       withNewHandle filePath $ \h -> do
         let payload = PayloadH h offset sz
-        pInfoMay <- checkUploadNeeded payload partNum pmap
-        maybe
-          (putObjectPart b o uploadId partNum [] payload)
-          return pInfoMay
+        putObjectPart b o uploadId partNum [] payload
 
 -- | Upload multipart object from conduit source sequentially
 sequentialMultipartUpload :: Bucket -> Object -> Maybe Int64
                           -> C.Producer Minio ByteString -> Minio ETag
 sequentialMultipartUpload b o sizeMay src = do
-  (uidMay, pmap) <- getExistingUpload b o
-
-  -- get a new upload id if needed.
-  uploadId <- maybe (newMultipartUpload b o []) return uidMay
+  -- get a new upload id.
+  uploadId <- newMultipartUpload b o []
 
   -- upload parts in loop
   let partSizes = selectPartSizes $ maybe maxObjectSize identity sizeMay
       (pnums, _, sizes) = List.unzip3 partSizes
   uploadedParts <- src
               C..| chunkBSConduit sizes
               C..| CL.map PayloadBS
-              C..| checkAndUpload uploadId pmap pnums
+              C..| checkAndUpload uploadId pnums
               C.$$ CC.sinkList
 
   -- complete multipart upload
   completeMultipartUpload b o uploadId uploadedParts
 
   where
-    checkAndUpload _ _ [] = return ()
-    checkAndUpload uid pmap (pn:pns) = do
+    checkAndUpload _ [] = return ()
```

Comment:
could you change the function to something like `uploadParts'`? the function is no longer checking anything.